### PR TITLE
[Merged by Bors] - TY-2553 configure stack ops upon creation

### PR DESCRIPTION
**Pull request by janpetschexain**
**References**

- [TY-2553]

**Summary**

- remove `configure()` method from stack `Ops`
- configure `BreakingNews` and `PersonalizedNews` upon creation

[TY-2553]: https://xainag.atlassian.net/browse/TY-2553?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by rustonaut — APPROVED** [review]
lgtm

**Review by rustonaut — CHANGES_REQUESTED** [review]
Sorry, I missed something.

markets do no longer need to be an option I think.

File: discovery_engine_core/core/src/stack/ops/personalized.rs [hunk 1]
```diff
@@ -40,28 +40,29 @@ use crate::{
 use super::Ops;
 
 /// Stack operations customized for personalized news items.
-#[derive(Default)]
 pub(crate) struct PersonalizedNews {
     client: Arc<Client>,
     markets: Option<Arc<RwLock<Vec<Market>>>>,
```
Comment (rustonaut) [inline discovery_engine_core/core/src/stack/ops/personalized.rs:45]:
```suggestion
    markets: Arc<RwLock<Vec<Market>>>,
```

File: discovery_engine_core/core/src/stack/ops/breaking.rs [hunk 2]
```diff
@@ -33,28 +33,29 @@ use crate::{
 use super::Ops;
 
 /// Stack operations customized for breaking news items.
-#[derive(Default)]
 pub(crate) struct BreakingNews {
     client: Arc<Client>,
     markets: Option<Arc<RwLock<Vec<Market>>>>,
```
Comment (rustonaut) [inline discovery_engine_core/core/src/stack/ops/breaking.rs:38]:
```suggestion
    markets: Arc<RwLock<Vec<Market>>>,
```

File: discovery_engine_core/core/src/stack/ops/breaking.rs [hunk 3]
```diff
@@ -33,28 +33,29 @@ use crate::{
...
     page_size: usize,
 }
 
+impl BreakingNews {
+    /// Creates a breaking news stack.
+    pub(crate) fn new(config: &EndpointConfig) -> Self {
+        Self {
+            client: Arc::new(Client::new(&config.api_key, &config.api_base_url)),
+            markets: Some(config.markets.clone()),
```
Comment (rustonaut) [inline discovery_engine_core/core/src/stack/ops/breaking.rs:47]:
```suggestion
            markets: config.markets.clone(),
```

File: discovery_engine_core/core/src/stack/ops/personalized.rs [hunk 4]
```diff
@@ -40,28 +40,29 @@ use crate::{
...
     page_size: usize,
 }
 
+impl PersonalizedNews {
+    /// Creates a personalized news stack.
+    pub(crate) fn new(config: &EndpointConfig) -> Self {
+        Self {
+            client: Arc::new(Client::new(&config.api_key, &config.api_base_url)),
+            markets: Some(config.markets.clone()),
```
Comment (rustonaut) [inline discovery_engine_core/core/src/stack/ops/personalized.rs:54]:
```suggestion
            markets: config.markets.clone(),
```

**Review by rustonaut — COMMENTED** [review]
Following, as well as removing the `if let Some(mekets)`'s but github doesn't let me make a change request there.

**Review by rustonaut — APPROVED** [review]
lgtm

**Comment by janpetschexain** [comment]
bors merge

**Pull request closed without merging**